lib/multi.c

```diff
@@ -2960,7 +2960,7 @@ static CURLMcode multi_timeout(struct Curl_multi *multi,
     /* splay the lowest to the bottom */
     multi->timetree = Curl_splay(tv_zero, multi->timetree);
 
-    if(Curl_splaycomparekeys(multi->timetree->key, now) > 0) {
+    if(Curl_splaycomparekeys(multi->timetree->key, now) >= 0) {
       /* some time left before expiration */
       timediff_t diff = Curl_timediff(multi->timetree->key, now);
       if(diff <= 0)
```

Comment:
This `if` branch was created to avoid busyloop. But it will never be executed without this patch. :-)

Reply:
I think you are mistaken. This block is to treat time differences greater than 0 but less than 1 millisecond as 1 millisecond. The only case where it times out immediately is if the time difference is 0. Your change would make it so that no longer happens.

Reply:
Even if the time difference is 0, I think it would be better if we add a little delta timeout instead of timing out immediately.

In some RTOS platform, well the NuttX simulator, curl is busylooping forever since the time difference is alway 0. The reason why this happans is because the system clock is updated in an idle thread, which will not be executed if curl is busylooping.

Reply:
I'm not sure what to do about that but I still think it is wrong not to expire such a timeout immediately.